scripts/deploy/functions.sh

```diff
@@ -97,17 +98,25 @@ function control_node() {
 
     # Only set {http,https,no}_proxy when a sidecar proxy container is not used.
     if [[ -v http_proxy ]]; then
+      proxy_variables_used=1
       docker_cmd+=( --env http_proxy )
     fi
 
     if [[ -v https_proxy ]]; then
+      proxy_variables_used=1
       docker_cmd+=( --env https_proxy )
     fi
 
     if [[ -v no_proxy ]]; then
+      proxy_variables_used=1
       docker_cmd+=( --env no_proxy )
     fi
 
+    if [[ $proxy_variables_used -eq 1 ]]; then
+      warn "HTTP proxy variables are used, but no transparent proxy started. Consider using --with-proxy."
```

Comment:
Is `--with-proxy` the correct option? It has a special sematic for kind: install redsocks in kind control plane. Is it different for aws/gcp scenarios?
```suggestion
      warn "HTTP proxy variables are used, but no transparent proxy started. Consider using --start-proxy."
```

Reply:
Fixed, also please re-review since other approach used